File: src/kyu_7/alphabet_symmetry/python/alphabet_symmetry_test.py
import codewars_test as test
from solution import solve

@test.describe('tests')
def tests():

    def do_test(strings, expected):
        actual = solve(strings.copy())
        test.assert_equals(actual, expected, f'for strings {strings}\n')

    @test.it("Basic tests")
    def basic_tests():
        do_test(["abode","ABc","xyzD"] ,[4,3,1])
        do_test(["abide","ABc","xyz"] ,[4,3,0])
        do_test(["IAMDEFANDJKL","thedefgh","xyzDEFghijabc"] ,[6,5,7])
        do_test(["encode","abc","xyzD","ABmD"] ,[1, 3, 1, 3])

File: src/kyu_7/alphabet_symmetry/cpp/alphabet_symmetry_test.cpp
#include <vector>

Describe(SampleTests)
{
    It(should_pass_sample_tests)
    {
        using v = std::vector<int>;
        Assert::That(solve({ "abode", "ABc", "xyzD" }), Equals(v{ 4, 3, 1 }), ExtraMessage(R"(Invalid result for [ "abode", "ABc", "xyzD" ])"));
        Assert::That(solve({ "abide", "ABc", "xyz" }), Equals(v{ 4, 3, 0 }), ExtraMessage(R"(Invalid result for [ "abide", "ABc", "xyz" ])"));
        Assert::That(solve({ "IAMDEFANDJKL", "thedefgh", "xyzDEFghijabc" }), Equals(v{ 6, 5, 7 }), ExtraMessage(R"(Invalid result for [ "IAMDEFANDJKL", "thedefgh", "xyzDEFghijabc" ])"));
        Assert::That(solve({ "encode", "abc", "xyzD", "ABmD" }), Equals(v{ 1, 3, 1, 3 }), ExtraMessage(R"(Invalid result for [ "encode", "abc", "xyzD", "ABmD" ])"));
        Assert::That(solve({}), Equals(v{}), ExtraMessage(R"(Invalid result for [ ])"));
    }
};